glk/osglk.c

```diff
@@ -408,6 +408,8 @@ void os_set_text_attr(int attr)
         glk_set_style(style_Subheader);
     else if (curattr & OS_ATTR_ITALIC)
         glk_set_style(style_Emphasized);
+    else if (curattr & OS_ATTR_MONOSP)
+        glk_set_style(style_Preformatted);
```

Comment:
We should be able to support all combinations of bold, italic, and monospaced, though we'll need to set up more stylehints.

Reply:
The combination of bold+italic is already supported, but the combination of monospace with anyhting else doesn't really map to any of the predefined Glk styles, so I left it out for now. (Aren't stylehints *technically* deprecated anyways? I know GlkOte doesn't usually support them.)

Reply:
We should be able to use user1 & 2 for this.

Zarf's GlkOte doesn't support them, but AsyncGlk does. So it would work fully in Parchment. And maybe I can persuade Zarf to switch to AsyncGlk for Lectrote too...

Reply:
While there isn't a distinct gestalt selector for stylehints, we could infer whether they're supported by testing for gestalt_GarglkText. If that's supported then use user1&2, otherwise only use style_Preformatted. But the question is which should take precedence... you have monospace take the lowest precedence, but I think other interpreters when unable to support every formatting combination have often made it have the highest priority, because getting ASCII diagrams right is seen as more important than showing bold or italics inside those ascii diagrams. What do you think?

Reply:
Yeah, makes sense.

Reply:
Okay... if `gestalt_GarglkText` is supported, `User1` will be monospace+bold and `User2` will be monospace+italic. (And anyone trying to do monospace+bold+italic will just have to come up with something else...)

Reply:
Actually this is the same situation as faced by ZVM, Bocfel etc. I think we could just copy it wholesale. (Including the stylehint resets, cause why not.)

https://github.com/curiousdannii/ifvms.js/blob/773fa3a3039f8edaff63181c57c2328547ec16ee/src/zvm/io.js#L360

It uses normal, subheader, emphasised, alert, performated, user1, user2, and alert.

Also rather than a big if-else-if, can we use `curattr` as the index into an array?

Reply:
Btw, `os_status_redraw` is also using user1, but I think we can replace that with calls to `garglk_set_reversevideo`. You don't need to do that as part of this PR.

Reply:
> While there isn't a distinct gestalt selector for stylehints, we could infer whether they're supported by testing for gestalt_GarglkText.

Wait, I just realised we can't infer this, because my RemGlk fork supports stylehints but not garglktext. 😢 

Options:

1. Wait for me to add support. Realistically will probably take at least 2-3 months.
2. Add a new gestalt just for stylehints.
3. Assume stylehints will work and make a custom CSS for Lectrote.

Reply:
Option 2 sounds reasonable (with option 1 being a longer-term goal, I guess?). Option 3 runs the risk of "monospace+anything else" combinations falling back to default formatting if the custom CSS is absent or doesn't work out for whatever reason.

Reply:
So I'll be adding in a second gestalt check, `gestalt_Stylehints` (is selector code `0x1101` available?) that is checked in addition to `gestalt_GarglkText`. If either of those is available, the interpreter will assume that basic font formatting style hints will be honored.

Reply:
Yep, that's exactly what I was going to suggest :)

Reply:
> Btw, os_status_redraw is also using user1, [...]

At least for Gargoyle, this is a non-issue. `os_status` changes the User1 style only for text grid windows, while we only need to alter the styles for text buffer windows.